Automattic/wp-calypso
# Plans: Adapt Jetpack Free card so its code can be re-used

Relates to `1200412004370260-as-1200532973279652`.

#### Changes proposed in this Pull Request

* Extract the core implementation of `JetpackFreeCard` and `JetpackFreeCardButton` into a new component, `ProductWithoutPrice`, that can be re-used for Jetpack CRM Free (or other free products in the future).
* Re-write `JetpackFreeCard` to use `ProductWithoutPrice`.
* Move `JetpackFreeCard` under the `my-sites/` hierarchy, since that's the only place it's used.
* Re-write the existing unit tests from `JetpackFreeCardButton` so that they test the `useJetpackFreeButtonProps` hook.

#### Testing instructions

* Evaluate styles, implementation, and unit test code, ensuring they're readable and intuitive.
* Verify that in all Calypso experiences, the "Get started for free" button on the Jetpack Free product card looks and behaves exactly the same as in production.
* Verify that all unit tests still pass.
